cmd/cloudx/client/handler.go

```diff
@@ -521,7 +521,23 @@ func (h *CommandHelper) GetProject(id string) (*cloud.Project, error) {
 		return nil, err
 	}
 
-	project, res, err := c.ProjectApi.GetProject(h.Ctx, id).Execute()
+	id := uuid.FromStringOrNil(projectOrSlug)
+	if id == uuid.Nil {
+		pjs, err := h.ListProjects()
+		if err != nil {
+			return nil, err
+		}
+		for _, pm := range pjs {
+			if pm.GetSlug() == projectOrSlug {
+				id = uuid.FromStringOrNil(pm.GetId())
+			}
+		}
+	}
+	if id == uuid.Nil {
+		return nil, errors.Errorf("No project selected! Please use the flag --project to specify one.\n")
+	}
```

Comment:
I would move this error message to the very top of this function and make the condition `projectOrSlug == ""`. At this point, the proper error should be similar to `no project found with slug %s, only slugs %v are known`. This makes it a lot easier for users to spot their mistakes.

Reply:
Thanks for the review! :) 

Should the output look like this then or should we only show the closest slug? 

```
no project found with slug romantic-kilby-g76if4751f, only slugs known are: 
romantic-kilby-g76if4751e
busy-jang-bwemahrjn5
busy-black-v1msdqmms0
ecstatic-swartz-ququpvs916
peaceful-kare-ugrr1tziby
stoic-dijkstra-4x7qznhyru
```

Reply:
I would just print the whole list. Maybe you can use the command listing the projects instead? It should at least have a type to show projects as nice tables.